results/cmd/api/main_test.go

```diff
@@ -437,3 +439,146 @@ func TestDeleteTaskRun(t *testing.T) {
+			Namespace: "demo",
+		},
+	}
+	t4 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "official",
+		},
+	}
+	testList := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	tt := []struct {
+		name   string
+		list   []*pb.TaskRun
+		filter string
+		expect []*pb.TaskRun
+	}{
+		{
+			name:   "test simple query",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1"`,
+			expect: []*pb.TaskRun{t1, t3, t4},
+		},
+		{
+			name:   "test logical &&",
+			list:   testList,
+			filter: `taskrun.api_version=="v1beta1" && taskrun.metadata.namespace=="demo"`,
+			expect: []*pb.TaskRun{t3, t4},
+		},
+		{
+			name:   "test logical ||",
+			list:   testList,
+			filter: `taskrun.metadata.name=="newtaskrun" || taskrun.metadata.namespace=="default"`,
+			expect: []*pb.TaskRun{t1, t2, t4, t5},
+		},
+		{
+			name:   "test startWith",
+			list:   testList,
+			filter: `taskrun.metadata.name.startsWith("new")`,
+			expect: []*pb.TaskRun{t4, t5},
+		},
+		{
+			name:   "test endWith",
+			list:   testList,
+			filter: `taskrun.metadata.name.endsWith("run")`,
+			expect: []*pb.TaskRun{t1, t3, t4, t5},
+		},
+		{
+			name:   "test empty filter",
+			list:   testList,
+			filter: "",
+			expect: testList,
+		},
+		{
+			name:   "test invalid field",
+			list:   testList,
+			filter: `task.name=="newtaskrun"`,
+		},
+		{
+			name:   "test invalid value type",
+			list:   testList,
+			filter: `taskrun.name==newtaskrun`,
+		},
+		{
+			name:   "test value not existing in the server record",
+			list:   testList,
+			filter: `taskrun.metadata.name=="notaskrun"`,
+			expect: nil,
+		},
+	}
+	for _, ts := range testList {
+		if _, err := srv.CreateTaskRunResult(nil, &pb.CreateTaskRunRequest{
+			TaskRun: ts,
+		}); err != nil {
+			t.Fatalf("could not create taskrun: %v", err)
+		}
+	}
+	// Test if we can find inserted taskruns
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			res, err := srv.ListTaskRunsResult(nil, &pb.ListTaskRunsRequest{
+				Filter: tc.filter,
+			})
+			if tc.name == "test invalid field" || tc.name == "test invalid value type" {
+				if err != nil && status.Code(err) != codes.NotFound {
+					t.Fatalf("failed on test %v: %v", status.Code(err), err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("could not list taskruns: %v", err)
+				}
+				taskList := res.GetItems()
+				sort.SliceStable(taskList, func(i, j int) bool {
+					return taskList[i].String() < taskList[j].String()
+				})
+				sort.SliceStable(tc.expect, func(i, j int) bool {
+					return tc.expect[i].String() < tc.expect[j].String()
+				})
+				if diff := cmp.Diff(taskList, tc.expect, protocmp.Transform()); diff != "" {
+					t.Fatalf("could not get the same taskrun: %v", diff)
+				}
```

Comment:
Nothing to do here. Just wanted to say thanks for keeping slice ordering in mind! :D

Reply:
Thank you!